### PR TITLE
Respect LiveQuery application configuration

### PR DESCRIPTION
**Pull request by wpride**
cross-request: https://github.com/dimagi/commcare-hq/pull/16859

cc @ctsims @millerdev

### CODE REVIEW

File: src/main/java/services/RestoreFactory.java [hunk 1]
```diff
@@ -414,4 +420,12 @@ public String getAsUsername() {
     public void setAsUsername(String asUsername) {
         this.asUsername = asUsername;
     }
+
+    public boolean isUseLiveQuery() {
```
Comment (proteusvacuum) [inline src/main/java/services/RestoreFactory.java]: does this get used anywhere?

File: src/main/java/services/RestoreFactory.java [hunk 2]
```diff
@@ -377,6 +378,11 @@ public String getRestoreUrl(boolean overwriteCache) {
             builder.append("&since=").append(syncToken);
         }
         builder.append("&device_id=").append(getSyncDeviceId());
+
+        if (useLiveQuery) {
+            builder.append("&livequery=true");
```
Comment (proteusvacuum) [inline src/main/java/services/RestoreFactory.java]: I think this should be `case_sync=livequery`

Comment (proteusvacuum) [inline src/main/java/services/RestoreFactory.java]: This should be case_sync=livequery

File: src/main/java/services/RestoreFactory.java (hunk 1 above)
Comment (wpride) [inline src/main/java/services/RestoreFactory.java]: Its Spring standard to have getters and setters for your all your class variables

File: src/main/java/services/RestoreFactory.java (hunk 2 above)
Comment (wpride) [inline src/main/java/services/RestoreFactory.java]: Fixed, thanks

File: src/main/java/services/RestoreFactory.java (hunk 1 above)
Comment (proteusvacuum) [inline src/main/java/services/RestoreFactory.java]: should you then use it on line 382?

**Pull request merged**